### PR TITLE
rate limit user requests

### PR DESCRIPTION
**Pull request by doodyparizada**

### CODE REVIEW

File: scripts/src/config.ts [hunk 1]
```diff
@@ -7,7 +7,9 @@ import { LogTarget } from "./logging";
 
 export interface LimitConfig {
 	hourly_registration: number;
-	minute_registration: number;
+	a: number;
```
Comment (liorama) [inline scripts/src/config.ts]: ?

File: scripts/src/utils/rate_limit.ts [hunk 2]
```diff
@@ -109,3 +109,18 @@ export async function assertRateLimitRegistration(app: Application) {
...
+	return await checkRateLimit(`user_reqs:${ userId }:${ duration.asSeconds() }`, duration, limit, TooManyUserRequests,
+		1, 2); // use lower number of buckets to reduce stress on redis as this is called on each user request
+}
+
+export async function assertRateLimitUserRequests(user: User) {
+	const app = (await Application.get(user.appId))!;
+
+	const limiters = [
+		await checkRateLimitUserRequests(app.id, app.config.limits.hourly_user_requests || 150, moment.duration({ hours: 1 })),
+		await checkRateLimitUserRequests(app.id, app.config.limits.minute_user_requests || 20, moment.duration({ minutes: 1 }))
```
Comment (liorama) [inline scripts/src/utils/rate_limit.ts]: I think you meant to pass the user ID for both calls to `checkRateLimitUserRequests`

**Review by liorama — CHANGES_REQUESTED** [review]

File: scripts/src/config.ts (hunk 1 above)
Comment (doodyparizada) [reply scripts/src/config.ts]: hahaha

File: scripts/src/utils/rate_limit.ts (hunk 2 above)
Comment (doodyparizada) [reply scripts/src/utils/rate_limit.ts]: you are right!

**Review by Doc999tor — APPROVED** [review]

**Pull request merged**